src/test/java/com/squareup/javapoet/MethodSpecTest.java

```diff
@@ -247,6 +254,26 @@ static void staticMethod() {
     }
   }
 
+  abstract static class AbstractClassWithPrivateAnnotation {
+
+    private @interface PrivateAnnotation{ }
+
+    abstract void foo(@PrivateAnnotation final String bar);
+  }
+
+  @Test public void overrideNotCopyParameterAnnotations() {
+    TypeElement abstractTypeElement = getElement(AbstractClassWithPrivateAnnotation.class);
+    ExecutableElement fooElement = ElementFilter.methodsIn(abstractTypeElement.getEnclosedElements()).get(0);
+    ClassName implClassName = ClassName.get("com.squareup.javapoet", "Impl");
+    TypeSpec type = TypeSpec.classBuilder(implClassName)
+            .superclass(abstractTypeElement.asType())
+            .addMethod(MethodSpec.overriding(fooElement).build())
+            .build();
+    JavaFileObject jfo = JavaFile.builder(implClassName.packageName, type).build().toJavaFileObject();
+    Compilation compilation = javac().compile(jfo);
+    CompilationSubject.assertThat(compilation).succeeded();
```

Comment:
Static import `assertThat()`?

Reply:
There is already a static import for `com.google.common.truth.Truth.assertThat` that is way more used than this one, so keeping that one as import and explicit usage of this here.

Reply:
You can overload static imports: https://github.com/square/javapoet/pull/772